Sources/MongoSwift/Operations/Operation.swift

```diff
@@ -21,14 +44,24 @@ internal struct DefaultOperationExecutor: OperationExecutor {
     internal func execute<T: Operation>(_ operation: T,
                                         client: MongoClient,
                                         session: ClientSession?) throws -> T.OperationResult {
-        // if a session was provided, use its underlying connection
-        if let session = session {
-            let conn = try session.getConnection(forUseWith: client)
+        switch operation.connectionUsage {
+        case .steals:
+            // don't return the connection to the pool, as the operation will handle it
```

Comment:
the meat of this case looks like a condensed version of the `.uses` case below, which is where I'm getting the smell for the suggestion above about reducing to two cases.